spec/models/manageiq/providers/native_operation_workflow_spec.rb

```diff
@@ -0,0 +1,186 @@
+describe ManageIQ::Providers::NativeOperationWorkflow do
+  context '.create_job' do
+    it 'leaves job waiting to start' do
+      options = {}
+      job = described_class.create_job(options)
+
+      expect(job.state).to eq('waiting_to_start')
+    end
+  end
+
+  context 'state transitions' do
+    before do
+      @job = described_class.create_job({})
+    end
+
+    context 'waiting_to_start' do
+      before do
+        @job.state = 'waiting_to_start'
+      end
+
+      it 'allows start signal' do
+        expect(@job).to receive(:start)
+        @job.signal(:start)
+      end
+
+      it 'doesn\'t allow poll_native_task signal' do
```

Comment:
Yeah, I started with `'` and wanted to keep it consistent...maybe should switch to `"` everywhere here